databuilder/databuilder/publisher/neo4j_csv_unwind_publisher.py

```diff
@@ -0,0 +1,501 @@
+        if encrypted_conf is not None:
+            driver_args['encrypted'] = encrypted_conf
+
+        self._driver = GraphDatabase.driver(**driver_args)
+        self._db_name = conf.get_string(NEO4J_DATABASE_NAME)
+
+        self._transaction_size = conf.get_int(NEO4J_TRANSACTION_SIZE)
+        self._confirm_rel_created = conf.get_bool(NEO4J_RELATIONSHIP_CREATION_CONFIRM)
+
+        # config is list of node label.
+        # When set, this list specifies a list of nodes that shouldn't be updated, if exists
+        self.create_only_nodes = set(conf.get_list(NEO4J_CREATE_ONLY_NODES, default=[]))
+        self.labels: Set[str] = set()
+        self.publish_tag: str = conf.get_string(JOB_PUBLISH_TAG)
+        self.additional_fields: Dict = conf.get(ADDITIONAL_FIELDS)
+        self.add_publisher_metadata: bool = conf.get_bool(ADD_PUBLISHER_METADATA)
+        self.publish_reverse_relationships: bool = conf.get_bool(PUBLISH_REVERSE_RELATIONSHIPS)
+        self._preserve_adhoc_ui_data = conf.get_bool(PRESERVE_ADHOC_UI_DATA)
+        if self.add_publisher_metadata and not self.publish_tag:
+            raise Exception(f'{JOB_PUBLISH_TAG} should not be empty')
+
+        LOGGER.info('Publishing Node csv files %s, and Relation CSV files %s',
+                    self._node_files,
+                    self._relation_files)
+
+    def _list_files(self, conf: ConfigTree, path_key: str) -> List[str]:
+        """
+        List files from directory
+        :param conf:
+        :param path_key:
+        :return: List of file paths
+        """
+        if path_key not in conf:
+            return []
+
+        path = conf.get_string(path_key)
+        return [join(path, f) for f in listdir(path) if isfile(join(path, f))]
+
+    def publish_impl(self) -> None:  # noqa: C901
+        """
+        Publishes Nodes first and then Relations
+        :return:
+        """
+
+        start = time.time()
+
+        LOGGER.info('Creating indices using Node files: %s', self._node_files)
+        for node_file in self._node_files:
+            self._create_indices(node_file=node_file)
+
+        LOGGER.info('Publishing Node files: %s', self._node_files)
+        while True:
+            try:
+                node_file = next(self._node_files_iter)
+                self._publish_node(node_file)
+            except StopIteration:
+                break
+
+        LOGGER.info('Publishing Relationship files: %s', self._relation_files)
+        while True:
+            try:
+                relation_file = next(self._relation_files_iter)
+                self._publish_relation(relation_file)
+            except StopIteration:
+                break
+
+        LOGGER.info('Committed total %i statements', self._count)
+
+        # TODO: Add statsd support
+        LOGGER.info('Successfully published. Elapsed: %i seconds', time.time() - start)
+
+    def get_scope(self) -> str:
+        return 'publisher.neo4j'
+
+    def _create_indices(self, node_file: str) -> None:
```

Comment:
idea: allow for list of pre and post-publish actions. and set this as a default pre-publish action